### PR TITLE
Display the time of the last statistics

### PR DESCRIPTION
**Pull request by stephdl**
Screenshot_2019-12-02 System - NS7 stephdl dynu net(1)

https://github.com/NethServer/dev/issues/5978

### CODE REVIEW

File: ui/public/i18n/language.json [hunk 1]
```diff
@@ -95,8 +95,20 @@
...
+        "Apr": "April",
+        "May": "May",
+        "Jun": "June",
+        "Jul": "July",
+        "Aug": "August",
+        "Sep": "September",
+        "Oct": "October",
+        "Nov": "November",
+        "Dec": "December",
+        "time_of_the_last_stats": "Last statistics"
```
Comment (gsanchietti) [inline ui/public/i18n/language.json]: I'd change to `Last statistics update`

File: api/read [hunk 2]
```diff
@@ -54,8 +54,19 @@ if ($cmd eq 'statistic') {
        next if ($true ne 'true'); # if not true, then next one 
        push @count, $jail;
    }
+    # if the first stats does not exists, use this stats date
```
Comment (gsanchietti) [inline api/read]: I think there is an easiest way to it: just throw the unix timestamp to the UI and use a Vue filter to format it.

@edospadoni do you think it should be changed or we can leave as is?

File: ui/public/i18n/language.json (hunk 1 above)
Comment (stephdl) [reply ui/public/i18n/language.json]: done

File: root/usr/libexec/nethserver/fail2ban-statistics [hunk 3]
```diff
@@ -74,6 +74,9 @@ foreach $key (keys %jails) {
     $data->{'TotalBannedIP'}->{$key} = $counterStat;
 }
 
+# we need to display the time of the last stats
+$data->{'date'} = time();
+
```
Comment (gsanchietti) [inline root/usr/libexec/nethserver/fail2ban-statistics]: Just remove the whole block.

File: api/read [hunk 4]
```diff
@@ -54,8 +54,12 @@ if ($cmd eq 'statistic') {
...
+    if ( ! $data->{"date"}) {
```
Comment (gsanchietti) [inline api/read]: The date of last statistics update doesn't need to be saved inside the JSON, just use the modification file time. See http://www.perl.com/doc/manual/html/pod/perlfunc/stat.html

Example:
```
my $epoch_timestamp = (stat($fh))[9];
```

Comment (stephdl) [reply api/read]: ok done

File: root/usr/libexec/nethserver/fail2ban-statistics (hunk 3 above)
Comment (stephdl) [reply root/usr/libexec/nethserver/fail2ban-statistics]: done

**Comment by stephdl** [comment]
I stat on `/var/lib/nethserver/fail2ban/fail2ban.json` if the file does not exist, fail2ban-status create it when it runs

**Pull request merged**